### PR TITLE
Update publication IDs to slugs

### PR DESCRIPTION
**Pull request by ckdesir**
## Overview

Migrate from publication IDs to slugs

## Changes Made

- Migrate from publication IDs to slugs
- Add animation transitions throughout screens
- Add OkHTTP to Retrofit

### CODE REVIEW

File: app/src/main/graphql/com/cornellappdev/volume/schema.graphqls [hunk 1]
```diff
@@ -67,26 +67,42 @@ type Publication {
...
+    offset: Float = 0
+    limit: Float = 25
+    publicationID: String!
+  ): [Article!]!
+  # Returns a list of <Articles> of size <limit> via the given list of <publicationIDs>. Results offsetted by <offset> >= 0.
+  getArticlesByPublicationIDs(
+    offset: Float = 0
+    limit: Float = 25
+    publicationIDs: [String!]!
+  ): [Article!]!
```
Comment (BenHarris4848) [inline app/src/main/graphql/com/cornellappdev/volume/schema.graphqls]: Why do we need getArticlesByPublicationID and getArticlesByPublicationSlug

File: app/src/main/java/com/cornellappdev/volume/data/repositories/PublicationRepository.kt [hunk 2]
```diff
@@ -23,17 +22,18 @@ class PublicationRepository @Inject constructor(private val networkApi: NetworkA
     suspend fun fetchAllPublications(): List<Publication> =
         networkApi.fetchAllPublications().dataAssertNoErrors.mapDataToPublications()
 
-    suspend fun fetchPublicationsByIDs(pubIDs: List<String>): List<Publication> =
-        networkApi.fetchPublicationsByIDs(pubIDs).dataAssertNoErrors.mapDataToPublications()
+    suspend fun fetchPublicationsByIDs(slugs: List<String>): List<Publication> =
```
Comment (BenHarris4848) [inline app/src/main/java/com/cornellappdev/volume/data/repositories/PublicationRepository.kt]: same thing here

**Review by BenHarris4848 — APPROVED** [review]
Everything looks good to me. I put it through as an approval because I'm sure if its necessarily a problem to have the extra getPublicationById methods, but if it is definitely remove those first.

**Review by emilyhu8 — APPROVED** [review]
Looks good thanks Chris!

File: app/src/main/java/com/cornellappdev/volume/data/repositories/PublicationRepository.kt (hunk 2 above)
Comment (ckdesir) [reply app/src/main/java/com/cornellappdev/volume/data/repositories/PublicationRepository.kt]: Oops forgot to change the name, thank you!

File: app/src/main/graphql/com/cornellappdev/volume/schema.graphqls (hunk 1 above)
Comment (ckdesir) [reply app/src/main/graphql/com/cornellappdev/volume/schema.graphqls]: Gonna delete getArticlesByPublicationID, should only be doing things with slugs!

Comment (ckdesir) [reply app/src/main/graphql/com/cornellappdev/volume/schema.graphqls]: I think the schema still technically has IDS but going to delete regardless

**Pull request merged**